src/software/firmware/includes/alarm.h

```diff
@@ -0,0 +1,78 @@
+/*
+    Copyright (C) 2020 Makers For Life
+*/
+/******************************************************************************
+ * @author Makers For Life
+ * @file alarm.h
+ * @brief Alarm structure
+ *****************************************************************************/
+
+#pragma once
+
+// INCLUDES =====================================================================
+
+// Externals
+
+#include <stdint.h>
+
+// Internals
+
+// ENUMS =====================================================================
+
+enum AlarmPriority { LOW, MEDIUM, HIGH };
+
+// CLASS =====================================================================
+
+class Alarm {
+ public:
+    /**
+     *
+     * @param p_priority Alarm priority
+     * @param p_code Alarm code
+     * @param p_detectionThreshold Alarm detection threshold
+     *
+     */
+    Alarm(AlarmPriority p_priority, uint8_t p_code, uint8_t p_detectionThreshold);
+
+    /**
+     * Get the alarm priority
+     */
+    AlarmPriority priotity();
```

Comment:
could be named `getPriority() ` or something like that to be more explicit on what it does.

Reply:
Good point. Fix in next commit.